contrib/google.golang.org/grpc.v12/grpc.go

```diff
@@ -75,7 +75,7 @@ func startServerSpanFromContext(ctx context.Context, method string, cfg *interce
 	// we will add the items in extraOpts
 	optsLocal := options.Copy(cfg.spanOpts...)
 	optsLocal = append(optsLocal, extraOpts...)
-	md, _ := metadata.FromContext(ctx) // nil is ok
+	md, _ := metadata.FromIncomingContext(ctx) // nil is ok
```

Comment:
on second thought, does this have any chance of breaking customers?

Reply:
Only if they are using `google.golang.org/grpc/metadata` version [`v1.2.1`](https://pkg.go.dev/google.golang.org/grpc@v1.2.1/metadata) (released in 2017) or below. Actually, our code is broken due the missing functions since `v1.6.0` (released in 2017 too).

Reply:
As I understand it from the terms of the [contrib README](https://github.com/DataDog/dd-trace-go/blob/main/contrib/README.md#usage), the name "grpc.v12" would imply this integration supports tracing grpc starting from **and including** v1.2.0. The [public docs](https://docs.datadoghq.com/tracing/trace_collection/compatibility/go/) imply the same, so there may be users of this integration with grpc v1.2.0 and v1.2.1. If these changes break compatibility for grpc v1.2.1 and below, that might be a problem. Feel free to DM me if you want to discuss further or if I've misunderstood something.

Reply:
I don't really understood ever why the contrib is `grpc.v12`, actually :) According to the contrib README, it should have been `grpc.v1` as the major version is 1, not 12. I don't find any other term or other contrib indicating that `v12` should be understood as `v1.2.0`.

Speaking of breaking compatibility, we have other contribs that broke compatibility between minor versions, like Fiber in #2275. Additionally, this contrib is planned to be removed in our next v2 major release because it's deprecated.

I think we can assume is safe to break this contrib's compatibility because `v1.2.1` and below were released in 2017, it's an important dependency with multiple CVEs, and because right now it's not possible to compile it with newer versions of the same library.

Reply:
I see. It may be worth updating the [public docs](https://docs.datadoghq.com/tracing/trace_collection/compatibility/go/#framework-compatibility) to specify that the grpcv12 contrib supports versions above 1.2.1. If you agree, I'm happy to help out and open the docs PR on your behalf.

Reply:
@mtoffl01 Totally. If you can update them, I'd appreciate it.